panic.go

```diff
@@ -0,0 +1,42 @@
+package slog
+
+import "fmt"
+
+// PanicError is a caught error from a panic() call
+type PanicError interface {
+	Error() string
+	Recovered() interface{}
+}
+
+// Recover attempts to recover from a panic
+func Recover() PanicError {
```

Comment:
recover() called outside defer is funky, might not do what we expect

Reply:
`slog.Recover()` is intended to be used inside a defer.

```go
defer func() {
   if err := slog.Recover(); err != nil {
      ...
   }
}()
```